### ISSUE TITLE
Finetune from AF parameters

### ISSUE THREAD

**Issue by C-de-Furina** [aqlaboratory/openfold#421]
Hi, I'm trying to train this model from an AF-multimer pretrained parameters. I notice that although it resumes from jax parameters, the loss is still pretty high and this model is far away from to be useful. Is this correct? My arguments are below.

--template_release_dates_cache_path=/home/xiaoguo.chen/openfold/mmcif_cache.json \
    --precision=bf16 \
    --gpus=2 --replace_sampler_ddp=True \
    --seed=42 \
    --deepspeed_config_path=/home/xiaoguo.chen/openfold/deepspeed_config.json \
    --checkpoint_every_epoch \
    --train_chain_data_cache_path=/home/xiaoguo.chen/openfold/chain_data_cache.json \
    --obsolete_pdbs_file_path=/home/xiaoguo.chen/xiaoguo.chen/AFdata/pdb_mmcif/obsolete.dat \
    --max_epochs=20 \
    --train_epoch_len=100\
    --config_preset="model_5_multimer_v3" \
    --num_nodes=1 \
    --resume_from_jax_params=/home/xiaoguo.chen/xiaoguo.chen/AFdata/params/params_model_5_multimer_v3.npz \
    --train_mmcif_data_cache_path=/home/xiaoguo.chen/openfold/mmcif_cache.json
![image](https://github.com/aqlaboratory/openfold/assets/164466256/c2f18ed6-f43f-4304-948b-851bf56b9a60)

**Comment by christinaflo** [comment]
Hi, so for multimer you only need the train_mmcif_data_cache_path, not the train_chain_data_cache_path, that is only required for monomer. Are you using precomputed alignments for training, I do not see it in the arguments listed? Also, could you share your curves over a few epochs?

**Comment by C-de-Furina** [comment]
> Hi, so for multimer you only need the train_mmcif_data_cache_path, not the train_chain_data_cache_path, that is only required for monomer. Are you using precomputed alignments for training, I do not see it in the arguments listed? Also, could you share your curves over a few epochs?

Thank you. Yes I use precomputed alignments. Mgnify_hits and uniref90_hits are downloaded and uniprots_hits are generated locally with .cif files.

As for the curve, I've seen many others show their loss curves but I'm not sure how to generate that, can you tell me how to do that?

**Comment by christinaflo** [comment]
I use [weights and biases](https://wandb.ai). If you make an account and pass the parameters --wandb, --experiment_name, --wandb_project, --wandb_entity to the training script it will log it for you, as well as all the validation metrics. It'll be helpful to see how those metrics look for your finetuning run.

**Comment by C-de-Furina** [comment]
> 我使用[权重和偏差](https://wandb.ai)。如果您创建一个帐户并将参数 --wandb、--experiment_name、--wandb_project、--wandb_entity 传递给训练脚本，它将为您记录它以及所有验证指标。了解这些指标对于您的微调运行有何影响将很有帮助。

I'm trying to generate it and I will post curves later. Btw, I downloaded some alignments from RODA, and then I used .cif files which are from AF dataset to generate entire MSA for lost chains and uniprot_hits for all chains. 

Which means, Assume an AB.cif from AF dataset includes two chains A and B. If there are A-mgnify_hits and A-uniref90_hits in RODA but no alignment for B,  then I generate A-uniprots_hits and all three B-hits.

 Is this feasible? I'm not sure if I can  combine them together.

**Comment by C-de-Furina** [comment]
> I use [weights and biases](https://wandb.ai). If you make an account and pass the parameters --wandb, --experiment_name, --wandb_project, --wandb_entity to the training script it will log it for you, as well as all the validation metrics. It'll be helpful to see how those metrics look for your finetuning run.

Hi, I've generated some curves.

![image](https://github.com/aqlaboratory/openfold/assets/164466256/a5551e17-eae3-4770-b154-0ca2182eaad0)
![image](https://github.com/aqlaboratory/openfold/assets/164466256/a39208a4-95d7-47a7-b44a-614c0e2c2998)
Compare to OF paper, lddt_ca and drmsd_ca are similer but all other losses are extremelly higher.

Further, I found that the parameters become chaos after several steps training. The first prediction is generated with AF parameters, the second one is with a checkpoint after three steps training, and the third one is after 100 steps.
![image](https://github.com/aqlaboratory/openfold/assets/164466256/df7d429a-40d9-48b7-b2aa-b9b994f51023)
![image](https://github.com/aqlaboratory/openfold/assets/164466256/c688f009-291a-44d7-a98e-185dd827b9a6)
![image](https://github.com/aqlaboratory/openfold/assets/164466256/047fcbee-f877-443c-8427-a79b1ed7f904)

It seems that the training conpletely breaks pretrained parameters instead of 'finetune', so I'm pretty comfused about this. Do I make any mistake in arguments? Thank you.

**Comment by vetmax7** [comment]
@C-de-Furina Hello!

Could you explain how did you use the checkpoint? As a extra tuning `--openfold_checkpoint_path checkpoint.ckpt` for alphafold model_1 (for example) or a model itself?

**Comment by C-de-Furina** [comment]
> @C-de-Furina Hello!
> 
> Could you explain how did you use the checkpoint? As a extra tuning for alphafold model_1 (for example) or a model itself?`--openfold_checkpoint_path checkpoint.ckpt`

If you mean by generating prediction, then I use these two arguments at the same time.

--config_preset model_5_multimer_v3
--openfold_checkpoint_path checkpoint.ckpt

**Comment by vetmax7** [comment]
> > @C-de-Furina Hello!
> > Could you explain how did you use the checkpoint? As a extra tuning for alphafold model_1 (for example) or a model itself?`--openfold_checkpoint_path checkpoint.ckpt`
> 
> If you mean how do I generate prediction, then I use these two arguments at the same time.
> 
> --config_preset model_5_multimer_v3 --openfold_checkpoint_path checkpoint.ckpt

Ok, thanks!

one more question: Have you tried to convert checkpoint *.ckpt file to model *.npz itself? and use your own model instead of model_5 of AlphaFold?

**Comment by C-de-Furina** [comment]
> > > @C-de-Furina Hello!
> > > Could you explain how did you use the checkpoint? As a extra tuning for alphafold model_1 (for example) or a model itself?`--openfold_checkpoint_path checkpoint.ckpt`
> > 
> > 
> > If you mean how do I generate prediction, then I use these two arguments at the same time.
> > --config_preset model_5_multimer_v3 --openfold_checkpoint_path checkpoint.ckpt
> 
> Ok, thanks!
> 
> one more question: Have you tried to convert checkpoint *.ckpt file to model *.npz itself? and use your own model instead of model_5 of AlphaFold?

No, actually I even don't know how to do what you said.

**Comment by vetmax7** [comment]
Ok, thanks!

**Comment by abhinavb22** [comment]
Did any of you encounter problem with deepspeed during multi-node/multi-GPU training like this - #486 ? Any suggestions would be really helpful
